app/src/main/java/mozilla/lockbox/store/ItemDetailStore.kt

```diff
@@ -201,4 +208,13 @@ open class ItemDetailStore(
         _originalItem.onNext(null.asOptional())
         sessionDisposable.clear()
     }
+
+    fun findSavedItem(): Observable<List<ServerPassword>> {
+        val item = _itemToSave.value?.value
+        return dataStore.filteredList(
+            item?.username,
+            item?.password,
+            item?.hostname
+        )
+    }
```

Comment:
I think I would recommend the method signature

```kotlin
fun findLatestSavedItem(): Observable<Optional<ServerPassword>>
```

In the case where `itemToSave.value?.value?.id` is non-empty, then return the item. Otherwise, do the filtering and return the only one, or null.

Reply:
If value is non-empty, we can nav to the item list then item detail. If it's null, then just nav to the item list.